### ISSUE TITLE
Downgrade process fails at ssh connection (MacOS/darwin)

### ISSUE THREAD

**Issue by tamarindmonkey** [LukeZGD/iOS-OTA-Downgrader#156]
First off, thank you for creating this tool. After looking through everything to find this issue, it was a massive undertaking writing all of the scripts involved.

## Issue
SSH failing on both USB and LAN attempt during downgrade attempt. ssh_config file needed a little rework for darwin
```
[Log] Copying stuff to device via SSH...
* Make sure OpenSSH/Dropbear is installed on the device and running!
* Dropbear is only needed for devices on iOS 10
* To make sure that SSH is successful, try these steps:
* Reinstall OpenSSH/Dropbear, reboot and rejailbreak, then reinstall them again

[Input] Enter the root password of your iOS device when prompted.
* Note that you will be prompted twice. Do not worry that your input is not visible, it is still being entered.
* The default password is "alpine"
waiting for connection
accepted connection, fd = 4
waiting for connection
Requesting connecion to USB device handle 5814 (serial: 9f39687503fc2358oo8i3s3f370ad0d89fee9cc4b), port 22
run_ctos_loop: fd = 4
run_stoc_loop: fd = 4
Warning: Permanently added '[127.0.0.1]:2222' (RSA) to the list of known hosts.
Received disconnect from 127.0.0.1 port 2222:2: Too many authentication failures for root from 127.0.0.1 port 49487 ssh2
Disconnected from 127.0.0.1 port 2222
lost connection
recv failed: Connection reset by peer
[Log] Cannot connect to device via USB SSH.
* Please try the steps above to make sure that SSH is successful
* Alternatively, you may use kDFUApp by tihmstar (from my repo, see "Troubleshooting" wiki page)
[Input] Press Enter/Return to continue anyway (or press Ctrl+C to cancel and try again)
recv failed: Connection reset by peer
[Log] Will try again with Wi-Fi SSH...
* Make sure that the device and your PC/Mac are on the same network!
* You can check for your device's IP Address in: Settings > WiFi/WLAN > tap the 'i' next to your network name
[Input] Enter the IP Address of your device: 192.168.1.49
[Log] Copying stuff to device via SSH...
Warning: Permanently added '192.168.1.49' (RSA) to the list of known hosts.
Received disconnect from 192.168.1.49 port 22:2: Too many authentication failures for root from 192.168.1.124 port 61329 ssh2
Disconnected from 192.168.1.49 port 22
lost connection

[Error] Cannot connect to device via SSH.
* Please try the steps above to make sure that SSH is successful
```

## Caused by
./resources/depends.sh gets the location for ssh, scp, and sets the switches/options for calling those commands from the script
https://github.com/LukeZGD/iOS-OTA-Downgrader/blob/d6770b7a2efab6b2e40fb1ef02499cd0f82f6a24/resources/depends.sh#L53-L55

./resources/ssh_config is missing a couple options needed by openSSH
https://github.com/LukeZGD/iOS-OTA-Downgrader/blob/d6770b7a2efab6b2e40fb1ef02499cd0f82f6a24/resources/ssh_config#L1-L3

## Demonstration of current behavior
Location of ssh under Catalina
```
catalina:iOS-OTA-Downgrader user$ which ssh
/usr/bin/ssh
```
The version details of OpenSSH installed by brew under Catalina as of 1/2/22, for posterity
```
catalina:iOS-OTA-Downgrader user$ /usr/bin/ssh -V
OpenSSH_8.1p1, LibreSSL 2.7.3
```
The contents of ./resources/ssh_config
```
catalina:iOS-OTA-Downgrader user$ cat resources/ssh_config
StrictHostKeyChecking no
UserKnownHostsFile tmp/known_hosts
HostKeyAlgorithms=+ssh-rsa
```
Piecing it all together
```
catalina:iOS-OTA-Downgrader user$ /usr/bin/ssh -F ./resources/ssh_config root@192.168.1.49
Failed to add the host to the list of known hosts (tmp/known_hosts).
Received disconnect from 192.168.1.49 port 22:2: Too many authentication failures for root from 192.168.1.124 port 61703 ssh2
Disconnected from 192.168.1.49 port 22
```

**Comment by tamarindmonkey** [comment]
With fixed ssh_config file from #157 
```
catalina:iOS-OTA-Downgrader user$ cat ./resources/ssh_config
Host *
    AddKeysToAgent yes
    IdentitiesOnly yes
    StrictHostKeyChecking no
    UserKnownHostsFile /dev/null
    HostKeyAlgorithms +ssh-rsa
```

Output
```
catalina:iOS-OTA-Downgrader user$ /usr/bin/ssh -F ./resources/ssh_config root@192.168.1.49
Warning: Permanently added '192.168.1.49' (RSA) to the list of known hosts.
root@192.168.1.49's password:
iPad:~ root# exit
logout
Connection to 192.168.1.49 closed.
```

**Comment by LukeZGD** [comment]
Fixed with #157 merged
